test/integ_tests/test_cost_tracking.py

```diff
@@ -94,7 +91,10 @@ def test_all_devices_price_search():
     tasks = {}
     for region in AwsDevice.REGIONS:
         s = AwsSession(boto3.Session(region_name=region))
-        for device in [device for device in devices if device.arn not in _RESERVATION_ONLY_DEVICES]:
+        # Skip devices with empty execution windows
+        for device in [
+            device for device in devices if len(device.properties.service.executionWindows) > 0
```

Comment:
Nit: From [PEP 8](https://peps.python.org/pep-0008/#programming-recommendations), ("For sequences, use the fact that empty sequences are false") this should be 
```suggestion
            device for device in devices if device.properties.service.executionWindows
```